### PR TITLE
Convert drivers to use I2CDevice interface (v3)

### PR DESCRIPTION
**Pull request by lucasdemarchi**
All I2C drivers are now converted. Tested:

  - ToshibaLED
  - Airspeed_I2C is tested by @tridge and probably with a pending fix on his branch
  - RCOutput_PCA9685
  - NavioLED

Would be nice if someone could test:

  - Changes to AP_RangeFinder
  - RCOutput_Bebop
  - AP_BattMonitor_SMBus
  - Display_SSD1306_I2C

### CODE REVIEW

**Comment by lucasdemarchi** [comment]
@mirkix could you test Display_SSD1306_I2C continues to work?
@jberaud could you test motor output in Bebop?

The other ones, I'm not sure who could test.

**Comment by lucasdemarchi** [comment]
ADS1115 changes are now done, I'm updating the PR message

**Comment by mirkix** [comment]
@lucasdemarchi I will check that tomorrow.

**Comment by tridge** [comment]
i've rebased my pr-disco branch on this and tested on a Disco:
https://github.com/tridge/ardupilot/tree/pr-disco-i2c-changes
it also adds I2CDevice for PX4, which I tested with a PX4 airspeed sensor.
Note that the smbus_quick() call does not work for airspeed. The 1 byte transfer write does.

**Comment by khancyr** [comment]
@lucasdemarchi will test maxbotix I2C  sonar monday (
public holiday in France). Ping me if I forget to report.

**Comment by lucasdemarchi** [comment]
> i've rebased my pr-disco branch on this and tested on a Disco:
> https://github.com/tridge/ardupilot/tree/pr-disco-i2c-changes
> it also adds I2CDevice for PX4, which I tested with a PX4 airspeed sensor.
> Note that the smbus_quick() call does not work for airspeed. The 1 byte transfer write does.

@tridge  ok, thanks for testing this.  It's very different from what the datasheet says. We may find problems later, but it's better to have it working like you did.

I'm removing the smbus_quick changes and squashing your fixup on the relevant commit here

**Comment by mirkix** [comment]
@lucasdemarchi Display_SSD1306_I2C is working. Tested today.

**Comment by lucasdemarchi** [comment]
Applied all the patches that were tested. The rest is still pending.

File: libraries/AP_RangeFinder/AP_RangeFinder_MaxsonarI2CXL.cpp [hunk 1]
```diff
@@ -22,25 +22,27 @@
...
    The constructor also initializes the rangefinder. Note that this
    constructor is not called until detect() returns true, so we
    already know that we should setup the rangefinder
 */
-AP_RangeFinder_MaxsonarI2CXL::AP_RangeFinder_MaxsonarI2CXL(RangeFinder &_ranger, uint8_t instance, RangeFinder::RangeFinder_State &_state) :
-    AP_RangeFinder_Backend(_ranger, instance, _state),
-    _dev(hal.i2c_mgr->get_device(0, AP_RANGE_FINDER_MAXSONARI2CXL_DEFAULT_ADDR))
+AP_RangeFinder_MaxsonarI2CXL::AP_RangeFinder_MaxsonarI2CXL(RangeFinder &_ranger, uint8_t instance, RangeFinder::RangeFinder_State &_state)
+    : AP_RangeFinder_Backend(_ranger, instance, _state)
+    , _dev(hal.i2c_mgr->get_device(1, AP_RANGE_FINDER_MAXSONARI2CXL_DEFAULT_ADDR))
```
Comment (khancyr) [inline libraries/AP_RangeFinder/AP_RangeFinder_MaxsonarI2CXL.cpp]: why change from 0 to 1 ?

Comment (lucasdemarchi) [inline libraries/AP_RangeFinder/AP_RangeFinder_MaxsonarI2CXL.cpp]: Changing to 0 was a mistake. The original version used hal.i2c which for majority of boards is /dev/i2c-1. A more correct fix would be to add it to HAL_Boards.h but I didn't think it was necessary for this driver

**Comment by lucasdemarchi** [comment]
I went ahead and applied this PR since it was blocking some other PRs I'm going to send. I did give it another round of review. If I missed anything we can fix up on top later.

**Pull request closed without merging**